chapter02/worksheets/solution2.1.7.8.scala

```diff
@@ -16,17 +9,21 @@
   from q to the corresponding strings from p.
   */
 
-// def reorder[A](p: Seq[A], q: Seq[Int]): Seq[A] = ???
-def myMapping[P,Q](p: Seq[P], q: Seq[Q]): Map[Q, P] = {
-  val erg = q.zip(p).toMap
+/** 
+  Exercise 2.1.7.8
+  Write the solution of Exercise 2.1.7.7 as a function with
+ type parameters P and Q instead of the fixed types Int and String. 
+
+  The return type of the function should be Map[P, Q]. Run some tests
+ using types P = Double and Q = Set[Boolean]. */
+
+def myMapping[P,Q](p: Seq[P], q: Seq[Q]): Map[P, Q] = {
+  val erg: Map[P, Q] = p.zip(q).toMap
   erg
   }
 
-val result = myMapping(Seq("asdf", "bsdf", "csdf", "dsdf"), Seq(1, 2, 3, 4))
+val result = myMapping(Seq( Set(true), Set(true, false), Set() ), Seq( 1.0, 2.0, 3.0))
 
-val expected = Map(1 -> "asdf", 2 -> "bsdf", 3 -> "csdf", 4 -> "dsdf")
-assert(result == expected)
+val expected: Map[scala.collection.immutable.Set[_ <: Boolean],Double] = Map(Set(true) -> 1.0, Set(true, false) -> 2.0, Set() -> 3.0)
```

Comment:
All is good except that you pasted this type `scala.collection.immutable.Set[_ <: Boolean]`. This is what the Scala compiler guessed. The actual more precise type is just `Set[Boolean]`. I suggest that you either simplify this or just omit the type annotation, it's not really necessary here.